DDInstall/DDInstall.psm1

```diff
@@ -0,0 +1,192 @@
+function Install-DDAgent
+{
+    param (
+        [Parameter()][ValidateSet('DotNet')][String[]]$WithAPMTracers,
+        [String] $AgentInstallURL,
+        [Parameter(Mandatory)] [String]$ApiKey,
+        [String]$Site,
+        [String]$Tags,
+        [String]$Hostname,
+        [String]$DDAgentUsername,
+        [String]$DDAgentPassword,
+        [String]$ApplicationDataDirectory,
+        [String]$ProjectLocation,
```

Comment:
This list of parameters comes from [our installer docs](https://docs.datadoghq.com/agent/basic_agent_usage/windows/?tab=commandline#installation-configuration-options).

There are other options we could choose to support (ie [these ones](https://docs.datadoghq.com/agent/basic_agent_usage/windows/?tab=commandline#more-agent-configuration-options)). We could also add these parameters explicitly, which has a couple advantages (ie properly documenting the supported options + adding support for auto-complete), or we could add a "catch all" parameter which would accept any number of additional options in the form `KEY=VALUE`. I'm open to either approach (or something different entirely!)

Reply:
I don't think supporting the additional installer options in the second link is necessary. Ideally customers should enable those features in the datadog.yaml file

Reply:
The script could check if `datadog.yaml` already exists and throw display a warning if the options are provided. A terminating error might be a bit much since then initial install and future installs would need different command lines.
`ApplicationDataDIrectory` is provided here and could be used for the path.
It's also in the registry at `HKLM\SOFTWARE\Datadog\Datadog Agent\ConfigRoot`

Reply:
> I don't think supporting the additional installer options in the second link is necessary. Ideally customers should enable those features in the datadog.yaml file

I disagree, at least for customers doing a first time install.  The reason we have the commadn line options is because editing the yaml for basic first time install is non-trivial, so we put the most basic of options on the command line.

Also the agent installer will ignore most, but  not all, of these options on upgrade.

I would suggest the following:
1) if it's a new install, or a reinstall but `datadog.yaml` doesn't have an api key, process everything as you have above (this is what the installer does)
2) if it's an upgrade/there's already a working config, don't pass _any_ options.  If the user passes any options, then either (a) put up yellow warning that they'll be ignored and continue or (b) fail the script and make them start it with no options.

either way with (2) they should be aware those options won't be accepted.

Suggest coordinating with #windows-agent to figure out which options are actually honored when, and as above either fail when they're going to be ignored, or at least print a warning.

Reply:
Derek, you'd have no way of knowing this, but in a separate Slack discussion while you were OOO, we discussed eventually wanting to remove these options from being accessible by the installer at all (i.e. installer editing the YAML) because customers find the behavior exceptionally confusing as to when the options are respected or not.

We'd prefer to steer customers away from passing them to the installer and towards using RC or your idea of an on-host GUI configurator. If we wanted them available via this module, then I'd suggest not passing them to the install (for reasons mentioned above) and instead have the module enable them directly (e.g. separate module commands that edit the YAML as opposed to the MSI installer).

All that to say, I think that any support for these options is outside the scope of the first version of the module, which is geared towards APM SSI. We can discuss if/how we want them supported in a later version.